Code/Bench/substruct_match.cpp

```diff
@@ -0,0 +1,114 @@
+#include <catch2/catch_all.hpp>
+#include <string>
+#include <vector>
+#include <istream>
+#include <filesystem>
+
+#include "bench_common.hpp"
+
+#include <GraphMol/ROMol.h>
+#include <GraphMol/SmilesParse/SmilesParse.h>
+#include <GraphMol/Substruct/SubstructMatch.h>
+
+using namespace RDKit;
+
+namespace bench_substruct_match {
+
+std::filesystem::path relative_to_rdbase(
+    const std::filesystem::path &relative) {
+  char *rdbase = std::getenv("RDBASE");
+  if (!rdbase) {
+    throw std::runtime_error("RDBASE environment variable not set");
+  }
+  std::filesystem::path path(rdbase);
+  path /= relative;
+  return path;
+}
+
+struct StructSmartsExample {
+  std::string smarts;
+};
+
+std::vector<StructSmartsExample> parse_smarts_examples(std::istream &in) {
+  std::vector<StructSmartsExample> examples;
+  std::string line;
+  while (std::getline(in, line)) {
+    if (line.starts_with("#") || line.starts_with(" ") || line.empty()) {
+      continue;
+    }
+    std::istringstream ss(line);
+    std::string smarts;
+    std::getline(ss, smarts, ' ');
+    examples.push_back({smarts});
+  }
+  return examples;
+}
+
+std::vector<StructSmartsExample> load_smarts_examples(
+    std::filesystem::path &path) {
+  std::ifstream file(path);
+  return parse_smarts_examples(file);
+}
+
+std::vector<ROMol> load_smarts_queries(const std::filesystem::path &relative) {
+  auto path = relative_to_rdbase(relative);
+  auto examples = load_smarts_examples(path);
+  std::vector<ROMol> queries;
+  for (const auto &example : examples) {
+    auto query = v2::SmilesParse::MolFromSmarts(example.smarts);
+    REQUIRE(query);
+    queries.push_back(*query);
```

Comment:
This is going to copy the molecule.
I'd suggest either adding an `std::move` here or changing the return value to be a vector of `unique_ptr`

Reply:
Will change to `queries.push_back(std::move(*query));`.